pkg/services/preferences/store/preferences_store.go

```diff
@@ -0,0 +1,100 @@
+package preferencesstore
+
+import (
+	"context"
+	"time"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type StoreImpl struct {
+	SqlStore sqlstore.Store
+	Cfg      *setting.Cfg
+}
+
+// TODO : Get and Set store methods
+type Store interface {
+	// TODO adjust the methods to Get/Set methods move logic to service
+	Get(context.Context, *models.GetPreferencesQuery) (*models.Preferences, error)
```

Comment:
Maybe Get / CMD etc can go into the ... service? and out of the global models? Since the Result in no longer part of it.

Reply:
@kylebrandt I am not sure I understand what do you mean

Reply:
In short, perhaps `models.GetPreferencesQuery` (and List.. Save.. etc) doesn't need to be in the models package. But I'm not sure if another location is better (e.g. here in the store, or perhaps in the service)

Reply:
These models could be under each service `store` package because it's under its domain. 

The only thing to check here is if these models are used inside `sqlstore` package to avoid dependency cycles.